stingray/multitaper.py

```diff
@@ -0,0 +1,670 @@
+        return jk_var
+
+    def rebin(self, df=None, f=None, method="mean"):
+        """Rebin the multitaper spectrum to a new frequency resolution ``df``.
+
+        Parameters
+        ----------
+        df: float
+            The new frequency resolution
+
+        Other Parameters
+        ----------------
+        f: float
+            the rebin factor. If specified, it substitutes df with ``f*self.df``
+
+        Returns
+        -------
+        bin_cs = :class:`Multitaper` (or one of its subclasses) object
+            The newly binned multitaper spectrum.
+            Note: this object will be of the same type as the object
+            that called this method. For example, if this method is called
+            from :class:`AveragedPowerspectrum`, it will return an object of class
+            :class:`AveragedPowerspectrum`, too.
+        """
+
+        if f is None and df is None:
+            raise ValueError('You need to specify at least one between f and '
+                             'df')
+        elif f is not None:
+            df = f * self.df
+
+        # rebin cross spectrum to new resolution
+        binfreq, binmtp, binerr, step_size = \
+            rebin_data(self.freq, self.power, df, self.power_err,
+                       method=method, dx=self.df)
+        # make an empty cross spectrum object
+        # note: syntax deliberate to work with subclass Powerspectrum
+        bin_mtp = copy.copy(self)
+
+        # store the binned periodogram in the new object
+        bin_mtp.freq = binfreq
+        bin_mtp.power = binmtp
+        bin_mtp.eigvals = self.eigvals
+        bin_mtp.df = df
+        bin_mtp.n = self.n
+        bin_mtp.norm = self.norm
+        bin_mtp.nphots = self.nphots
+        bin_mtp.power_err = binerr
+
+        if hasattr(self, 'unnorm_power'):
+            _, binpower_unnorm, _, _ = \
+                rebin_data(self.freq, self.unnorm_power, df,
+                           method=method, dx=self.df)
+
+            bin_mtp.unnorm_power = binpower_unnorm
+
+        if hasattr(self, 'multitaper_norm_power'):
+            _, bin_multitaper_norm_power, _, _ = \
+                rebin_data(self.freq, self.multitaper_norm_power, df,
+                           method=method, dx=self.df)
+
+            bin_mtp.multitaper_norm_power = bin_multitaper_norm_power
+
+        if hasattr(self, 'jk_var_deg_freedom'):
+            _, bin_jk_var_deg_freedom, _, _ = \
+                rebin_data(self.freq, self.jk_var_deg_freedom, df,
+                           method=method, dx=self.df)
+
+            bin_mtp.jk_var_deg_freedom = bin_jk_var_deg_freedom
+
+        bin_mtp.m = np.rint(step_size * self.m)
+
+        return bin_mtp
+
+    def compute_rms(self, min_freq, max_freq, white_noise_offset):
+        return Powerspectrum.compute_rms(self, min_freq, max_freq,
+                                         white_noise_offset=white_noise_offset)
+
+    def classical_significances(self, threshold, trial_correction):
```

Comment:
I'm not sure that the classical significances matter here. It is used in the original Power spectrum because the spectrum itself is distributed according to the chi squared distribution, and you calculate the significances that way. This is not the case here. Isn't there a specific significance calculation for the Multitaper?